lab_3_ann_retriever/main.py

```diff
@@ -220,6 +285,23 @@ def _calculate_tf_idf(self, document: list[str]) -> Vector | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(document, list) or not document:
+            return None
+        vector = [0.0]*len(self._vocabulary)
+        for token in document:
+            tf = calculate_tf(self._vocabulary, document)
+            idf = self._idf_values
+            if not isinstance(tf, dict):
+                return None
+            tf_idf = calculate_tf_idf(tf, idf)
+            if not isinstance(tf_idf, dict):
+                return None
+            token_index = self._token2ind[token]
+            vector[token_index] = tf_idf[token]
+        tf_idf_vector = tuple(vector)
+        if tf_idf_vector is None or not isinstance(tf_idf_vector, tuple) or not tf_idf_vector:
```

Comment:
Strange check. A line above you create a tuple and then check if the variable is a tuple

Reply:
done